lib/facter/custom_facts/util/directory_loader.rb

```diff
@@ -93,7 +93,12 @@ def load_searched_facts(collection, searched_facts, weight)
           if data == false
             log.warn "Could not interpret fact file #{fact.file}"
           elsif (data == {}) || data.nil?
-            log.debug("Fact file #{fact.file} was parsed but no key=>value data was returned")
+            log.debug(
+              "Structured data fact file #{fact.file} was parsed but was either empty or an invalid filetype "\
+                '(valid filetypes are .yaml, .json, and .txt).'
+            )
+          elsif !data.is_a?(Hash)
+            log.error("Structured data fact file #{fact.file} was parsed but no key=>value data was returned.")
```

Comment:
Does this change how facter resolves `ini` style facts, for example [facts.txt in facter3](https://github.com/puppetlabs/facter/blob/3.x/lib/tests/fixtures/facts/external/text/facts.txt) and the corresponding [test](https://github.com/puppetlabs/facter/blob/6d7ffc6efdfbc3b1fc79311cdeb4581ac2098d9c/lib/tests/facts/external/text_resolver.cc#L21-L33)?

Reply:
It seems like both Facter 3 and 4 behave the same in that instance.

The last Facter 3 release:
```
root@cockier-panther:~# /opt/puppetlabs/bin/facter --version
3.14.24 (commit 91ed8a2de5c9d686345859fe12ea2914415758f0)
root@cockier-panther:~# cat /tmp/facts.d/test.txt
txt_fact1=value1
txt_fact2=
not a fact
TXT_Fact4=value2
root@cockier-panther:~# /opt/puppetlabs/bin/facter --external-dir  /tmp/facts.d/ txt_fact1
value1
root@cockier-panther:~# /opt/puppetlabs/bin/facter --external-dir  /tmp/facts.d/ txt_fact2

root@cockier-panther:~# /opt/puppetlabs/bin/facter --external-dir  /tmp/facts.d/ TXT_Fact4

```
Versus this branch:
```
$ git describe
4.5.0-7-g47a729a11
$ cat /tmp/facts.d/test.txt
txt_fact1=value1
txt_fact2=
not a fact
TXT_Fact4=value2
$ bundle exec facter --external-dir  /tmp/facts.d/ txt_fact1
value1
$ bundle exec facter --external-dir  /tmp/facts.d/ txt_fact2

$ bundle exec facter --external-dir  /tmp/facts.d/ TXT_Fact4

```